polygon/bor/bordb/prune.go

```diff
@@ -17,27 +17,321 @@
+	}
+
+	return err
+}
+
+func UnwindEventNums(tx kv.RwTx, unwindPoint uint64) error {
+	c, err := tx.RwCursor(kv.BorEventNums)
+	if err != nil {
+		return err
+	}
+
+	defer c.Close()
+	var blockNumBuf [8]byte
+	binary.BigEndian.PutUint64(blockNumBuf[:], unwindPoint+1)
+	var k []byte
+	for k, _, err = c.Seek(blockNumBuf[:]); err == nil && k != nil; k, _, err = c.Next() {
+		if err = c.DeleteCurrent(); err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+func UnwindEventProcessedBlocks(tx kv.RwTx, unwindPoint uint64) error {
+	c, err := tx.RwCursor(kv.BorEventProcessedBlocks)
+	if err != nil {
+		return err
+	}
+
+	defer c.Close()
+	var blockNumBuf [8]byte
+	binary.BigEndian.PutUint64(blockNumBuf[:], unwindPoint+1)
+	var k []byte
+	for k, _, err = c.Seek(blockNumBuf[:]); err == nil && k != nil; k, _, err = c.Next() {
+		if err = c.DeleteCurrent(); err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+func UnwindSpans(ctx context.Context, heimdallStore heimdall.Store, tx kv.RwTx, unwindPoint uint64) error {
+	_, err := heimdallStore.Spans().(interface {
+		WithTx(kv.Tx) heimdall.EntityStore[*heimdall.Span]
+	}).WithTx(tx).DeleteFromBlockNum(ctx, unwindPoint)
+
+	return err
+}
+
+func UnwindSpanBlockProducerSelections(ctx context.Context, heimdallStore heimdall.Store, tx kv.RwTx, unwindPoint uint64) error {
+	_, err := heimdallStore.SpanBlockProducerSelections().(interface {
+		WithTx(kv.Tx) heimdall.EntityStore[*heimdall.SpanBlockProducerSelection]
+	}).WithTx(tx).DeleteFromBlockNum(ctx, unwindPoint)
+
+	return err
+}
+
+func UnwindCheckpoints(ctx context.Context, heimdallStore heimdall.Store, tx kv.RwTx, unwindPoint uint64) error {
+	_, err := heimdallStore.Checkpoints().(interface {
+		WithTx(kv.Tx) heimdall.EntityStore[*heimdall.Checkpoint]
+	}).WithTx(tx).DeleteFromBlockNum(ctx, unwindPoint)
+
+	return err
+}
+
+func UnwindMilestones(ctx context.Context, heimdallStore heimdall.Store, tx kv.RwTx, unwindPoint uint64) error {
+	_, err := heimdallStore.Milestones().(interface {
+		WithTx(kv.Tx) heimdall.EntityStore[*heimdall.Milestone]
+	}).WithTx(tx).DeleteFromBlockNum(ctx, unwindPoint)
+
+	return err
+}
+
+func PruneHeimdall(ctx context.Context, heimdallStore heimdall.Store, bridgeStore bridge.Store, tx kv.RwTx, blocksTo uint64, blocksDeleteLimit int) (int, error) {
+	var deleted int
+
+	eventsDeleted, err := PruneEvents(tx, blocksTo, blocksDeleteLimit)
+
+	if err != nil {
+		return eventsDeleted, err
+	}
+
+	deleted = eventsDeleted
+
+	spansDeleted, err := heimdallStore.Spans().(interface {
+		WithTx(kv.Tx) heimdall.EntityStore[*heimdall.Span]
+	}).WithTx(tx).DeleteToBlockNum(ctx, blocksTo, blocksDeleteLimit)
+
+	if spansDeleted > deleted {
+		deleted = spansDeleted
+	}
+
+	if err != nil {
+		return deleted, err
+	}
+
+	spansDeleted, err = heimdallStore.SpanBlockProducerSelections().(interface {
+		WithTx(kv.Tx) heimdall.EntityStore[*heimdall.SpanBlockProducerSelection]
+	}).WithTx(tx).DeleteToBlockNum(ctx, blocksTo, blocksDeleteLimit)
```

Comment:
How do we handle the "standalone case" for pruning when Heimdall and Bridge data is in separate Heimdall/Bridge  DBs? In that case the "tx" needs to be instantiated from the right DB (Heimdall/Bridge) instead of from the Chaindata DB